@amimodel/compileC.m

```diff
@@ -4,113 +4,25 @@ function compileC(this)
     % Return values:
     %  this: model definition object @type amimodel
     
-    sundials_path = fullfile(this.wrap_path,'sundials');
-    sundials_ver = '2.7.0';
-    
-    ssparse_path = fullfile(this.wrap_path,'SuiteSparse');
-    ssparse_ver = '4.5.3';
-    
-    lapack_path = fullfile(this.wrap_path,'lapack-3.5.0'); % currently not used, lapack implementation still needs to be done
-    lapack_ver = '3.5.0';
-    
-    % compile directory
-    if(~exist(fullfile(this.wrap_path,'models',mexext), 'dir'))
-        mkdir(fullfile(this.wrap_path,'models',mexext))
-    end
-    
-    [del_sundials, del_ssparse, del_lapack] = checkVersions(this, sundials_ver, ssparse_ver, lapack_ver);
-    
-    sources_sundials = getSourcesSundials();
-    sourcesstr_sundials = '';
-    for j=1:length(sources_sundials)
-        sourcesstr_sundials = strcat(sourcesstr_sundials,' "',fullfile(sundials_path,sources_sundials{j}),'"');
-    end
-    
-    sources_ssparse = getSourcesSSparse();
-    sourcesstr_ssparse = '';
-    for j=1:length(sources_ssparse)
-        sourcesstr_ssparse = strcat(sourcesstr_ssparse,' "', fullfile(ssparse_path,sources_ssparse{j}),'"');
-    end
-    
-    sourcesstr_funs = '';
-    for j=1:length(this.funs)
-        sourcesstr_funs = strcat(sourcesstr_funs,...
-            ' "', fullfile(this.wrap_path,'models',this.modelname,[this.modelname '_' this.funs{j} '.c']),'"');
-    end
-    
-    objects_sundials = getObjectsSundials();
-    if(ispc)
-        objects_sundials = strrep(objects_sundials, '.o', '.obj');
-    end
-    
-    objects_ssparse = getObjectsSSparse();
-    if(ispc)
-        objects_ssparse = strrep(objects_ssparse, '.o', '.obj');
-    end
-    
-    
-    % assemble objectsstr
-    objectsstr = '';
-    for j=1:length(objects_sundials)
-        objectsstr = strcat(objectsstr,' "',fullfile(this.wrap_path,'models',mexext,objects_sundials{j}),'"');
-    end
-    
-    for j=1:length(objects_ssparse)
-        objectsstr = strcat(objectsstr,' "',fullfile(this.wrap_path,'models',mexext,objects_ssparse{j}),'"');
-    end
-    
     if(ispc)
         o_suffix = '.obj';
     else
         o_suffix = '.o';
     end
     
+    amiciSourcePath = fullfile(this.wrap_path,'src');
+    modelSourceFolder = fullfile(this.wrap_path,'models',this.modelname);
+
+    [objectsstr, includesstr] = compileAMICIDependencies(this.wrap_path, o_suffix);
+        
+    includesstr = strcat(includesstr,' -I"', modelSourceFolder, '"');
```

Comment:
I propose to have the debug and optimized object files in two different directories, and link them depending on `model.debug`.